_episodes/l1-06-releases_tags.md

```diff
@@ -3,19 +3,355 @@ title: "Code versions, releases and tags"
+- "Know how to tag a given commit"
+- "Understand how to give your software meaningful version numbers with semantic versioning"
+- "Know how to push your tags and publish a release of your software"
 keypoints:
-- TODO
+- "A version of your code with a release number (e.g. v13.4.2) is referred to as a *release*"
+- "A version of your code represented by a commit hash (e.g. 047e4fe) is just referred to as a *commit*"
+- "Publishing a release can be a good way to bundle features and ensure your users use a specific version of your code"
+- "`git tag` allows you to give a commit a human-readable name, such as a version number"
+- "Semantic versioning is a common convention for conveying to your users what a new version number means"
+- "Tags need to be explicitly pushed to remotes with `git push --tags`"
+- "You can use a tag as the basis for a release on GitHub"
 ---
 
-## TITLE
+## Background: To release or not to release?
+
+All of you will already be familiar with the concept of software versioning. Often when
+you download a piece of software from its website it'll tell you that it's `v13.4.2` (or
+whatever) that you're downloading.
+
+Releasing software with an explicit version number like this is a common practice and
+one that you may eventually consider for some of your own projects. We will show you how
+to do this using Git and GitHub. Even if you never end up making a release yourself,
+rest assured that sooner or later you will have to work with a repository which uses
+releases like this, so it's important to understand the concepts at least.
+
+The first question you might have is:
+
+### Why do I need a "version" for my software? Isn't Git tracking the version anyway?
+
+Sort of. The problem here is that the word "version" can mean several different things
+in the context of Git. Ordinarily, when people talk about a particular version of a
+piece of software, they mean a version with a particular release number, such as
+`v13.4.2`. However, in another sense, each commit in your repository represents a
+different version of the code and can be represented by a unique commit hash (e.g.
+`047e4fe`).
+
+To avoid confusion, people usually refer to the first kind of version as a "release" and
+the second as a "commit", which is what we'll do here.
+
+### When *don't* I need to create releases for my software?
+
+You probably don't need to create releases for your software if:
+
+- The code never changes
+- You're the only developer and the only user (e.g. it's some analysis scripts that only
+  you use)
+- (OR there are only a few of you and you all just use the latest commit etc.)
+- The software is simple and doesn't change much from one commit to the next
+
+In this case, if you just need to clarify which commit you're working from (e.g. to a
+colleague), remember that you can always obtain the current commit hash like so:
+
+~~~sh
+$ git rev-parse --short HEAD
+~~~
+{: .commands}
+~~~
+047e4fe
+~~~
+{: .output}
```

Comment:
I would consider moving the "When *don't* you need..." section below the "When should I consider..." section. It's subjective, but for me it reads better.

Reply:
Good idea